tests/formatconverter/formatconvertertest.cpp

```diff
@@ -54,155 +53,185 @@ using std::string;
+
+    msg = wxT("input: '") + input +
+        wxT("', result (UTF-8): '") + result +
+        wxT("', expected: '") + expectedUtf8 + wxT("'");
+    CHECK(result == expectedUtf8);
+#endif // wxUSE_UNICODE_UTF8
+
+#if !wxUSE_UTF8_LOCALE_ONLY
+    result = (const wchar_t*)wxFormatString(input);
+
+#if defined(__WINDOWS__) && \
+    !defined(__CYGWIN__) && \
+    !defined(__MINGW32__)
+    wxString expectedWchar(expectedWcharWindows);
+#else
+    wxString expectedWchar(expectedWcharUnix);
+#endif
+
+    msg = wxT("input: '") + input +
+        wxT("', result (wchar_t): '") + result +
+        wxT("', expected: '") + expectedWchar + wxT("'");
+    CHECK(result == expectedWchar);
+#endif // !wxUSE_UTF8_LOCALE_ONLY
+}
+
+void doTest(const char* input, const char* expectedScanf,
+    const char* expectedUtf8,
+    const char* expectedWcharUnix,
+    const char* expectedWcharWindows)
+{
+    static const wxChar* flag_width[] =
+    { wxT(""), wxT("*"), wxT("10"), wxT("-*"), wxT("-10"), nullptr };
+    static const wxChar* precision[] =
+    { wxT(""), wxT(".*"), wxT(".10"), nullptr };
+    static const wxChar* empty[] =
+    { wxT(""), nullptr };
+
+    // no precision for %c or %C
+    const wxChar** precs = wxTolower(input[wxStrlen(input) - 1]) == wxT('c') ?
+        empty : precision;
+
+    wxString fmt(wxT("%"));
+
+    // try the test for a variety of combinations of flag, width and precision
+    for (const wxChar** prec = precs; *prec; prec++)
+        for (const wxChar** width = flag_width; *width; width++)
+            check(fmt + *width + *prec + input,
+                fmt + *width + *prec + expectedScanf,
+                fmt + *width + *prec + expectedUtf8,
+                fmt + *width + *prec + expectedWcharUnix,
+                fmt + *width + *prec + expectedWcharWindows);
+}
+
+
+TEST_CASE("format_d", "[formatconverter]")
 {
     doTest("d", "d", "d", "d", "d");
-    CPPUNIT_ASSERT(wxString::Format(wxT("%d"), 255) == wxT("255"));
-    CPPUNIT_ASSERT(wxString::Format(wxT("%05d"), 255) == wxT("00255"));
-    CPPUNIT_ASSERT(wxString::Format(wxT("% 5d"), 255) == wxT("  255"));
-    CPPUNIT_ASSERT(wxString::Format(wxT("% 5d"), -255) == wxT(" -255"));
-    CPPUNIT_ASSERT(wxString::Format(wxT("%-5d"), -255) == wxT("-255 "));
-    CPPUNIT_ASSERT(wxString::Format(wxT("%+5d"), 255) == wxT(" +255"));
-    CPPUNIT_ASSERT(wxString::Format(wxT("%*d"), 5, 255) == wxT("  255"));
+    CHECK(wxString::Format(wxT("%d"), 255) == wxT("255"));
+    CHECK(wxString::Format(wxT("%05d"), 255) == wxT("00255"));
+    CHECK(wxString::Format(wxT("% 5d"), 255) == wxT("  255"));
+    CHECK(wxString::Format(wxT("% 5d"), -255) == wxT(" -255"));
+    CHECK(wxString::Format(wxT("%-5d"), -255) == wxT("-255 "));
+    CHECK(wxString::Format(wxT("%+5d"), 255) == wxT(" +255"));
+    CHECK(wxString::Format(wxT("%*d"), 5, 255) == wxT("  255"));
 }
 
-void FormatConverterTestCase::format_hd()
+TEST_CASE("format_hd", "[formatconverter]")
```

Comment:
For consistency I'd use

```suggestion
TEST_CASE("Format::hd", "[formatconverter]")
```

here and below.